src/libexpr/print-options.hh

```diff
@@ -8,6 +8,29 @@
 
 namespace nix {
 
+/**
+ * How errors should be handled when printing values.
+ */
+enum class ErrorPrintBehavior {
+    /**
+     * Print the first line of the error in brackets: `«error: oh no!»`
+     */
+    Print,
+    /**
+     * Throw the error to the code that attempted to print the value, instead
+     * of suppressing it it.
+     */
+    Throw,
+    /**
+     * Only throw the error if encountered at the top level of the expression.
+     *
+     * This will cause expressions like `builtins.throw "uh oh!"` to throw
+     * errors, but will print attribute sets and other nested structures
+     * containing values that error (like `nixpkgs`) normally.
+     */
+    ThrowTopLevel,
+};
```

Comment:
What's the reason for this (hidden) configurability?

Reply:
Do the comments not make it clear enough? Let me know how I can improve it. Output snipped for brevity, but basically if the whole expression errors we want to print the error message normally, but in nested structures like attribute sets or lists we want to show error messages for specific attributes (to get a good overview for structures like `nixpkgs`):

```
nix-repl> pkgs.pythonPackages.APScheduler
error:
       … in the condition of the assert statement
         at /nix/store/wmj6j3afrvqxsjpad47nrad1vrwmykrg-source/lib/customisation.nix:268:17
       … in the left operand of the OR (||) operator
         at /nix/store/wmj6j3afrvqxsjpad47nrad1vrwmykrg-source/pkgs/development/interpreters/python/passthrufun.nix:28:45

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: Package ‘python-2.7.18.7’ in /nix/store/wmj6j3afrvqxsjpad47nrad1vrwmykrg-source/pkgs/development/interpreters/python/cpython/2.7/default.nix:335 is marked as insecure, refusing to evaluate.


       Known issues:
        - Python 2.7 has reached its end of life after 2020-01-01. See https://www.python.org/doc/sunset-python-2/.

       You can install it anyway by allowing this package, using the
       following methods:

       a) To temporarily allow all insecure packages, you can use an environment
          variable for a single invocation of the nix tools:

            $ export NIXPKGS_ALLOW_INSECURE=1

          Note: When using `nix shell`, `nix build`, `nix develop`, etc with a flake,
                then pass `--impure` in order to allow use of environment variables.
...
nix-repl> pkgs.pythonPackages
{
  APScheduler = «error: Package ‘python-2.7.18.7’ in /nix/store/wmj6j3afrvqxsjpad47nrad1vrwmykrg-source/pkgs/development/interpreters/python/cpython/2.7/default.nix:335 is marked as insecure, refusing to evaluate.»;
  BTrees = «error: Package ‘python-2.7.18.7’ in /nix/store/wmj6j3afrvqxsjpad47nrad1vrwmykrg-source/pkgs/development/interpreters/python/cpython/2.7/default.nix:335 is marked as insecure, refusing to evaluate.»;
  Babel = «error: Package ‘python-2.7.18.7’ in /nix/store/wmj6j3afrvqxsjpad47nrad1vrwmykrg-source/pkgs/development/interpreters/python/cpython/2.7/default.nix:335 is marked as insecure, refusing to evaluate.»;
error: interrupted by the user
```

Reply:
Oh sorry, I missed the bit about this not being repl-specific. Forget about my comment.

(I guess we're never using the `Throw` variant, but it doesn't harm to have it either)